### PR TITLE
Support for pulling SIF images directly from OCI registry

### PR DESCRIPTION
**Pull request by EmmEff**
Implements support for pulling SIF images directly from OCI registry without using Cloud Library as an intermediary.

### CODE REVIEW

File: client/downloader.go [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+
+	c.Logger.Logf("size: %d, parts: %d, streams: %d, partsize: %d", size, parts, spec.Concurrency, spec.PartSize)
+
+	g, ctx := errgroup.WithContext(ctx)
+
+	// Allocate channel for file part requests
+	ch := make(chan filePartDescriptor, parts)
+
+	// Create download part workers
+	for n := uint(0); n < defaultStreams; n++ {
```
Comment (dtrudg) [inline client/downloader.go:74]: Does this mean we are always using 4 streams, despite the fact we are logging a spec.Concurrency above?

**Review by dtrudg — COMMENTED** [review]
Have done an initial read through. I'm a bit unclear if we are actually honoring spec.Concurrency now, despite logging it?

File: client/downloader.go (hunk 1 above)
Comment (EmmEff) [reply client/downloader.go:74]: Yes it does... I'll fix it :)

**Review by dtrudg — APPROVED** [review]
LGTM

**Review by tri-adam — CHANGES_REQUESTED** [review]
Looks good functionally!

[Gorelease](https://pkg.go.dev/golang.org/x/exp/cmd/gorelease) is picking up several new symbols that have been added to the public API, which we probably want to avoid given the compatibility guarantee implied by the fact this is a 1.x module? I've made a suggestion for each.

Thanks!

**Review by tri-adam — APPROVED** [review]

**Pull request merged**